firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -226,93 +349,85 @@ public void onClick(DialogInterface dialogInterface, int i) {
     return alertDialog;
   }
 
-  /** Signs in the App Distribution tester. Presents the tester with a Google sign in UI */
-  @NonNull
-  public Task<Void> signInTester() {
-    if (signInTaskCompletionSource != null && !signInTaskCompletionSource.getTask().isComplete()) {
-      signInCancellationSource.cancel();
-    }
-
-    signInCancellationSource = new CancellationTokenSource();
-    signInTaskCompletionSource = new TaskCompletionSource<>(signInCancellationSource.getToken());
-
-    Context context = firebaseApp.getApplicationContext();
-    AlertDialog alertDialog = getSignInAlertDialog(context);
-    alertDialog.show();
-
-    return signInTaskCompletionSource.getTask();
-  }
-
-  private void setSignInTaskCompletionError(FirebaseAppDistributionException e) {
-    if (signInTaskCompletionSource != null && !signInTaskCompletionSource.getTask().isComplete()) {
-      signInTaskCompletionSource.setException(e);
-    }
-  }
-
-  /** Returns true if the App Distribution tester is signed in */
-  @NonNull
-  public boolean isTesterSignedIn() {
-    return false;
+  private OnSuccessListener<String> getFidGenerationOnSuccessListener(Context context) {
+    return new OnSuccessListener<String>() {
+      @Override
+      public void onSuccess(String fid) {
+        Uri uri =
+            Uri.parse(
+                String.format(
+                    "https://appdistribution.firebase.google.com/pub/apps/%s/installations/%s/buildalerts?appName=%s&packageName=%s",
+                    firebaseApp.getOptions().getApplicationId(),
+                    fid,
+                    getApplicationName(context),
+                    context.getPackageName()));
+        openSignInFlowInBrowser(uri);
+      }
+    };
   }
 
-  /** Signs out the App Distribution tester */
-  public void signOutTester() {}
+  private void handleLatestReleaseFromClient(
+      String fid, String appId, String apiKey, String authToken) {
+    checkForUpdateExecutor.execute(
+        new Runnable() {
+          @Override
+          public void run() {
+            try {
+              AppDistributionRelease latestRelease =
+                  firebaseAppDistributionTesterApiClient.fetchLatestRelease(
+                      fid, appId, apiKey, authToken);
+
+              Context context = firebaseApp.getApplicationContext();
+              long currentInstalledVersionCode = getInstalledAppVersionCode(context);
+
+              if (Long.parseLong(latestRelease.getBuildVersion()) <= currentInstalledVersionCode) {
```

Comment:
In the future, when we have signature verification for AABs and APKs, we'll want this comparison to be strictly less than. For now, I think we should add a TODO reminding us to change this when signature verification is in place